packages/docusaurus-theme-classic/src/theme/ThemedImage/index.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import type {Props} from '@theme/ThemedImage';
+import useThemeContext from '@theme/hooks/useThemeContext';
+
+function ThemedImage(props: Props): JSX.Element {
+  const {isDarkTheme} = useThemeContext();
+  const {src, darkSrc, lightSrc, alt = '', ...propsRest} = props;
+
+  if (darkSrc && isDarkTheme) {
```

Comment:
Let's not duplicate essentially the same code, let's use a ternary expression to determine the desired image src?

And why is there a prop just `src`? In this case need use native `<img>` element, not `ThemedImage`.

Reply:
> Let's not duplicate essentially the same code, let's use a ternary expression to determine the desired image src?

I choose the standard conditional structure for the redability.

> And why is there a prop just `src`? In this case need use native `<img>` element, not `ThemedImage`.

Because of TS types inheritance - `& ComponentProps<'img'>;`  - and props passing. It also serves as fallback, if user provide `src` instead of `lightSrc` to the themed image.